apps/web/src/actions/profile.ts

```diff
@@ -26,16 +28,20 @@ import { revalidatePath } from "next/cache";
 import { cookies } from "next/headers";
 import { Argon2id } from "oslo/password";
 import "server-only";
+import { waitUntil } from "@vercel/functions";
+import type { Session } from "lucia";
 import type { z } from "zod";
 
 export async function updateBaseInformationUsername(data: z.infer<typeof baseInformationSchema>) {
+    let session: Session | undefined = undefined;
+    const userDataToLog = "private";
     try {
-        const { user, session } = await getActionSession();
+        session = (await getActionSession())?.session as Session;
```

Comment:
Warum hier der compiler force zu as Session?

Reply:
War durch den Compiler in meiner IDE vorgegeben, ansonsten lässt der die Zuweisung zur Variable nicht zu

Reply:
```tsx
    let session: Session | null = null;
    const userDataToLog = "private";
    try {
        session = (await getActionSession())?.session;
        ....
     } catch (err) {
         .....
     }
```

nicht undefined sondern null.

Reply:
Es klappt, ist halt nicht 100% sauber. Die Funktion getActionSession hat eine Session oder null als Return hat.

Du überschreibst den Compiler und sagst du bist definitiv eine Session und nicht null.

Am Ende passiert nichts wegen der Prüfung, wenn alle damit fein sind würde ich das okay finden.

Reply:
Habs eben umgesetzt.